src/algorithms/digi/PhotoMultiplierHitDigi.cc

```diff
@@ -63,23 +63,25 @@ void eicrecon::PhotoMultiplierHitDigi::AlgorithmChangeRun() {
 //------------------------
 // AlgorithmProcess
 //------------------------
-std::vector<edm4eic::RawTrackerHit*>
-eicrecon::PhotoMultiplierHitDigi::AlgorithmProcess(std::vector<const edm4hep::SimTrackerHit*>& sim_hits) {
-
+std::vector<edm4eic::MCRecoTrackerHitAssociation*> eicrecon::PhotoMultiplierHitDigi::AlgorithmProcess(
```

Comment:
Does it make sense to wait a few days until we can return both the RawTrackerHit collection AND the (optional) MCRecoTrackerHitAssocation collection? #578

Reply:
Sure. The noise hits from #555 won't have associated `SimTrackerHit` collections, so in that case the output `RawTrackerHit` collection would have more hits than the output association collection. Or do you think we should fill the association collection with noise hits too, but with empty `SimTrackerHit`s? I can make either case work in the IRT wrapper algorithm (which has "cheat modes" to access the true photon information, if desired).

Reply:
No, I think it's fine (correct, even) to not have associations for noise hits. Associations don't need to provide an association for every entry, but every association must point to a valid entry in each collection.

Reply:
I've now done this, but in a separate PR which targets merging into this one, so that the additional changes are clear. See https://github.com/eic/EICrecon/pull/612, which has been tested locally and works fine.